vllm/model_executor/models/baichuan.py

```diff
@@ -351,8 +352,42 @@ def load_weights(self,
             ("gate_up_proj", "up_proj", 1),
         ]
         params_dict = dict(self.named_parameters())
-        for name, loaded_weight in hf_model_weights_iterator(
-                model_name_or_path, cache_dir, load_format, revision):
+        weight_names = []
+        for name in params_dict:
+            if "gate_up_proj" in name:
+                weight_names.append(name.replace("gate_up_proj", "gate_proj"))
+                weight_names.append(name.replace("gate_up_proj", "up_proj"))
+            else:
+                weight_names.append(name)
+        weight_names = sorted(weight_names)
+
+        def weights_iterator(
+            use_distributed_loading: bool
+        ) -> Iterator[Tuple[str, Optional[torch.Tensor], Optional[int]]]:
+            if use_distributed_loading:
+                local_state_dict, weights_map = \
+                    hf_model_weights_distributed_loader(
+                        model_name_or_path, weight_names,
+                        cache_dir, load_format, revision)
+                for name in weight_names:
+                    weight_owner = weights_map[name]
+                    loaded_weight = local_state_dict[name].to("cuda") \
+                        if weight_owner == get_tensor_model_parallel_rank() \
+                        else None
+                    yield name, loaded_weight, weight_owner
+            else:
+                for name, loaded_weight in hf_model_weights_iterator(
+                        model_name_or_path, cache_dir, load_format, revision):
+                    yield name, loaded_weight, None
```

Comment:
Why does this need to be defined within the model file instead of as a separate helper?

Reply:
Fixed